plugins/module_utils/network/sonic/config/sflow/sflow.py

```diff
@@ -0,0 +1,522 @@
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.errno)
+            result['changed'] = True
+        result['commands'] = commands
+        changed_sflow_facts = self.get_sflow_facts()
+
+        result['before'] = existing_sflow_facts
+        if result['changed']:
+            result['after'] = changed_sflow_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_sflow_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_sflow_facts
+
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A tuple
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration, and REST requests that do it
+        """
+        commands = []
+        requests = []
+        want = self.validate_normalize_config(want)
+        state = self._module.params['state']
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        elif state == 'overridden':
+            commands, requests = self._state_overridden(want, have)
+        else:
+            commands, requests = self._state_replaced(want, have)
+        return commands, requests
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A tuple of lists
+        :returns: A list of what commands and state necessary to migrate the current configuration
+                  to the desired configuration, and a list of requests needed to make changes
+        """
+        commands = []
+        requests = []
+
+        replaced_config = get_replaced_config(want, have, test_keys=self.sflow_diff_test_keys)
+        if replaced_config:
+            requests.extend(self.get_deleted_requests(replaced_config, have))
+            commands.extend(update_states(replaced_config, "deleted"))
+            add_commands = want
+        else:
+            diff = get_diff(want, have, test_keys=self.sflow_diff_test_keys)
+            add_commands = diff
+            self.prep_merge_diff(add_commands, want)
```

Comment:
Looking at it in detail seems like it was more useful before using replaced_config and checking if that existed. Before that change it was deleting the whole interface and then using the diff to add the new portions of each interface. Some interfaces passed in could container some fields that were the same and the diff would not normally contain those. they had to be added back in to difference.

Reply:
Thanks for explaining.

Based on this information, it sounds like the call to prep_merge_diff can safely be removed from this "else" block. (Since the "else" block only handles the case in which no deletion is occurring, the original reason for doing this no longer exists and the "add_commands" diff should be correct with no further modifications.)